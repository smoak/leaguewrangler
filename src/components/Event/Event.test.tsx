import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import EventWithStyles, { Event } from '.';

describe('Event', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(
      <Event
        classes={{ card: 'card', icon: 'icon' }}
        title="foo"
        startTime={new Date(1557029700000)}
        location="Somewhere USA"
        teamPhotoUrl="https://picsum.photos/48"
      />
    );
  });

  it('renders', () => {
    expect(component).toMatchSnapshot();
  });
});

describe('EventWithStyles', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(
      <EventWithStyles
        title="foo"
        startTime={new Date(1557029700000)}
        location="Somewhere USA"
        teamPhotoUrl="https://picsum.photos/48"
      />
    );
  });

  it('renders', () => {
    expect(component).toMatchSnapshot();
  });
});
