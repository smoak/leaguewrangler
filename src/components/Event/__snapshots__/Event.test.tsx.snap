// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders as expected 1`] = `
<WithStyles(ForwardRef(Card))
  className="card"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <WithStyles(ForwardRef(Avatar))
        aria-label="teamPhoto"
        src="https://picsum.photos/48"
      />
    }
    subheader={
      <t
        ago={false}
        calendar={false}
        date={2019-05-05T04:15:00.000Z}
        decimal={false}
        element={null}
        filter={[Function]}
        format="LLLL"
        fromNow={false}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      />
    }
    title="foo"
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="p"
    >
      Somewhere USA
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(EventActions)
      eventId={1}
      rsvpStatus="NONE"
      teamId={1}
    />
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`Event when the event has no startTime renders as expected 1`] = `
<WithStyles(ForwardRef(Card))
  className="card"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <WithStyles(ForwardRef(Avatar))
        aria-label="teamPhoto"
        src="https://new.url"
      />
    }
    subheader={null}
    title="foo"
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="p"
    >
      Somewhere USA
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(EventActions)
      eventId={1}
      rsvpStatus="NONE"
      teamId={1}
    />
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`Event when the event has no team photo url renders as expected 1`] = `
<WithStyles(ForwardRef(Card))
  className="card"
>
  <WithStyles(ForwardRef(CardHeader))
    avatar={
      <WithStyles(ForwardRef(Avatar))
        aria-label="teamPhoto"
        src=""
      />
    }
    subheader={
      <t
        ago={false}
        calendar={false}
        date={2019-05-05T04:15:00.000Z}
        decimal={false}
        element={null}
        filter={[Function]}
        format="LLLL"
        fromNow={false}
        interval={60000}
        local={false}
        onChange={[Function]}
        titleFormat=""
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
        withTitle={false}
      />
    }
    title="foo"
  />
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      component="p"
    >
      Somewhere USA
    </WithStyles(ForwardRef(Typography))>
  </WithStyles(ForwardRef(CardContent))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(EventActions)
      eventId={1}
      rsvpStatus="NONE"
      teamId={1}
    />
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;

exports[`EventWithStyles renders 1`] = `
<Event
  classes={
    Object {
      "card": "Event-card-1",
    }
  }
  eventId={1}
  location="Somewhere USA"
  rsvpStatus="NONE"
  startTime={2019-05-05T04:15:00.000Z}
  teamId={1}
  teamPhotoUrl="https://picsum.photos/48"
  title="foo"
/>
`;
